visivo/models/base/eval_string.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any, List
+import re
+from visivo.models.base.context_string import ContextString
+from visivo.parsers.evaluator import evaluate_expression, SUPPORTED_NUMPY_FUNCTIONS
+
+INLINE_CONTEXT_STRING_REGEX = r"\${\s*[\(a-zA-Z0-9\s'\"\-_\\.\]\[)]+?\s*}"
+EVAL_STRING_REGEX = r"^>{(.*)}$"
+
+"""
+EvalString represents a string that contains expressions to be evaluated.
+
+This class is used for handling strings that may contain references or expressions
+enclosed in '>{' and '}' delimiters. It provides functionality to validate the
+string format and extract embedded ContextString instances.
+
+Available injected methods:
+    - any_test_failed() -> bool
```

Comment:
I wonder if it makes sense to nest these in a `run` or `test` object that gets updated when you run visivo cli commands. Then you could reference it like `test.any_test_failed`

Reply:
I am not sure I followed that.